java/maven/src/main/java/net/sf/mmm/code/java/maven/impl/MavenBridgeImpl.java

```diff
@@ -213,25 +213,33 @@ public Model readEffectiveModelFromLocation(File location, boolean fallback) {
 
   private Properties resolveProperties(Properties properties, File projectBaseDir) {
 
-    try {
-      File mvnDir = new File(projectBaseDir, ".mvn");
-      if (mvnDir.isDirectory()) {
-        File mvnConfig = new File(mvnDir, "maven.config");
-        if (mvnConfig.isFile()) {
-          properties = new Properties(properties);
-          List<String> lines = Files.readAllLines(mvnConfig.toPath());
-          for (String line : lines) {
-            resolveProperties(properties, line.trim());
+    if (projectBaseDir.getParentFile() == null) {
+      return properties;
+    }
+
+    boolean pomFileExists = new File(projectBaseDir, POM_XML).exists();
+    boolean parentPomFileExists = new File(projectBaseDir.getParentFile(), POM_XML).exists();
```

Comment:
actually parent POM may not always be `../pom.xml`.

Reply:
Actually I don't know whether this will fix our issue, but maybe one of them. There might remain the other issue (needs to be tested) to calculate the effective model.
You are right, but I was not able to find what "project's top level directory" does actually mean exactly. For me it was not bound to the parent pom, which might even be packed again in some sub-directory. Therefore, I was pointing to the top-level pom.xml, which is executed, no matter whether this one might have a parent which even is referenced in a sub-directory.